File: EasyProblems/Python/twosum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        indexed_nums = dict()
        for i in range(0, len(nums)):
            match = target - nums[i]
            if match in indexed_nums.keys():
                return [indexed_nums[match], i]
            
            indexed_nums[nums[i]] = i
        return -1

File: EasyProblems/Scala/twosum.scala
object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        var numToIndex = Map.empty[Int, Int]
        
        for ((num, i) <- nums.zipWithIndex) {
            numToIndex.get(target - num) match {
                case Some(x) => return Array(x, i)
                case None => numToIndex += (num -> i)
            }
        }
        
        Array(-1,-1)
    }
}